examples/glue_benchmark/glue.py

```diff
@@ -0,0 +1,234 @@
+        label = x["label"]
+        return (features, label)
+
+    if task_name == "ax":
+        # AX is trained and evaluated on MNLI, and has its own test split.
+        train_ds, validation_ds = tfds.load(
+            "glue/mnli",
+            split=["train", "validation_matched"],
+        )
+        test_ds = tfds.load(
+            "glue/ax",
+            split="test",
+        )
+    else:
+        train_ds, test_ds, validation_ds = tfds.load(
+            f"glue/{task_name}",
+            split=["train", "test" + test_suffix, "validation" + test_suffix],
+        )
+    train_ds = train_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    test_ds = test_ds.map(split_features, num_parallel_calls=tf.data.AUTOTUNE)
+    validation_ds = validation_ds.map(
+        split_features, num_parallel_calls=tf.data.AUTOTUNE
+    )
+    return train_ds, test_ds, validation_ds
+
+
+def preprocess_data(preprocess_fn, dataset):
+    """Run `proprocess_fn` on input dataset then batch & prefetch."""
+    return (
+        dataset.map(preprocess_fn)
+        .batch(FLAGS.batch_size)
+        .prefetch(tf.data.AUTOTUNE)
+    )
+
+
+def generate_submission_files(finetuning_model, test_ds):
+    """Generate GLUE leaderboard submission files."""
+    filenames = {
+        "cola": "CoLA.tsv",
+        "sst2": "SST-2.tsv",
+        "mrpc": "MRPC.tsv",
+        "qqp": "QQP.tsv",
+        "stsb": "STS-B.tsv",
+        "mnli_matched": "MNLI-m.tsv",
+        "mnli_mismatched": "MNLI-mm.tsv",
+        "qnli": "QNLI.tsv",
+        "rte": "RTE.tsv",
+        "wnli": "WNLI.tsv",
+        "ax": "AX.tsv",
+    }
+
+    labelnames = {
+        "mnli_matched": ["entailment", "neutral", "contradiction"],
+        "mnli_mismatched": ["entailment", "neutral", "contradiction"],
+        "ax": ["entailment", "neutral", "contradiction"],
+        "qnli": ["entailment", "not_entailment"],
+        "rte": ["entailment", "not_entailment"],
+    }
+    if not os.path.exists(FLAGS.submission_directory):
+        os.makedirs(FLAGS.submission_directory)
+    filename = FLAGS.submission_directory + "/" + filenames[FLAGS.task_name]
+
+    labelname = labelnames.get(FLAGS.task_name)
+    with tf.io.gfile.GFile(filename, "w") as f:
+        # GLUE requires a format of index + tab + prediction.
+        writer = csv.writer(f, delimiter="\t")
+        # Write the required headline for GLUE.
+        writer.writerow(["index", "prediction"])
+        predictions = finetuning_model.predict(test_ds)
+        predictions = np.argmax(predictions, -1)
+        for idx, pred in enumerate(predictions):
+            if labelname:
+                pred_value = labelname[int(pred)]
+            else:
+                pred_value = pred
+            writer.writerow([idx, pred_value])
+
+
+def main(_):
+    train_ds, test_ds, val_ds = load_data(FLAGS.task_name)
+
+    # ----- Custom code block starts -----
+    bert_model = keras_nlp.models.Bert.from_preset("bert_tiny_uncased_en")
```

Comment:
Maybe add a line after this.

Users should change this block to implement the preprocessing required by the model.